Add friendly-shell, friendly-shell-command, friendly-remote-shell

p3r7 opened the pull request:
Cleaner attempt of https://github.com/melpa/melpa/pull/6834.

### Brief summary of what the package does

More user-friendly shell APIs for Emacs.

3 packages:
 - `friendly-shell-command.el` for single shell commands ([blog post](https://www.eigenbahn.com/2020/01/19/painless-emacs-shell-commands))
 - `friendly-shell.el` for interactive shells ([blog post](https://www.eigenbahn.com/2020/01/21/painless-emacs-interactive-shells))
 - `friendly-remote-shell.el` for interactive remote shells

### Direct link to the package repository

https://github.com/p3r7/friendly-shell

### Your association with the package

Maintainer.

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented on recipes/friendly-remote-shell:
```diff
@@ -0,0 +1,4 @@
+(friendly-remote-shell :repo "p3r7/friendly-remote-shell"
```
This repository doesn't seem to exist at the moment. :)

p3r7 replied:
See my comment bellow.

riscy commented on recipes/friendly-shell-command:
```diff
@@ -0,0 +1,4 @@
+(friendly-shell-command :repo "p3r7/friendly-shell-command"
```
This repository doesn't exist too.

p3r7 replied:
See my comment bellow.

p3r7 commented:
> Sorry it's taken a while to get around to this second go round.

No problem. I understand the amount of work this represent :wink: 

> Usually when there are multiple recipes, it's much preferred to open three different PR's given three different recipes.

Oh my bad, I made a mistake : the 3 packages are in fact hosted on the same repo (as they are very closely related) which exists. I guess I blindly did a search / replace all :facepalm: 

I've fixed the repo reference in the recipes with https://github.com/melpa/melpa/pull/6851/commits/5e8f9070aa858b7ca121844e934b653606abfdbd.

If originally (attempted at least) to follow this guideline from [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org) : `If the packages are very closely related and maintained in the same repository (see below for why it might not be a good idea), you might open only one pull request for them all; in this case, keep adding one recipe per commit.`.

How does that look now?

riscy commented:
Ah you're right about CONTRIBUTING.org :)  I might see if there's a possibility to amend that, because if the packages depend on one-another then it's harder to test if their dependencies are accessible from MELPA (an important check).  Also the yes/no decision whether to merge becomes easier when it's just one package.

That's not a blocker, but fyi.  In the meantime:

byte-compile-file (using Emacs 26.3):
```
friendly-shell-command.el:24:1:Error: Cannot open load file: No such file or directory, dash
```
I think a `Package-Requires` is needed on dash in this case.

Minor stuff I see:
- proccess -> process
- necessayr -> necessary
- according -> according
- DIPLAY -> DISPLAY

p3r7 commented:
- [x] Added missing dash dependency in the `Package-Requires` header.
 - [x] Fixed typos (I'm again sorry for those :facepalm:) 
 - [x] Completed documentation with recently added features

> if the packages depend on one-another then it's harder to test if their dependencies are accessible from MELPA (an important check)

So should I split this PR into several? Just tell me.

riscy commented:
No, it's fine. :)

riscy commented:
Okay, these look good to me!

The one (non-blocking) suggestion I might be to remove the underline in your commentary (or at least be aware it'll look like this (note the extra space) in `finder-commentary`:
```elisp
Commentary:
 -----------

For detailed instructions, please look at the README.md
```
and, worse, I'm not sure whether that underline will appear in the [MELPA index's "description"](  That will be helpful to people looking at the "description" part of the MELPA index ([example](https://melpa.org/#/bind-key)) section for the packages.  I'd also recommend giving a link to that README for ease of use (so somebody can just use find-file-at-point to open the link) or -- even better -- offer a few quick details about the package too.

I think that's not worth blocking on, though.  These look cool.

p3r7 commented:
Indeed, I've added a more comprehensive package header doc, which was lacking.

Thanks for the tip about the underline.
Indeed it wasn't working so nice. I guess I originally took inspiration from another package's source but did not investigate how they would work with other packages.
I've removed them from all my other packages.

Thanks again for all your feedback!
